File: eecs268/2020spring/video-lectures/2020.04.01.mazeWalker-part-2/MazeWalker.cpp


//sketch of walk


//public facing walk
bool MazeWalker::walk()
{
	return( walk(m_startRow, m_startCol) );	
}


bool MazeWalker::walk(int row, int col)
{
	//mark current position
	mark(row, col);
	
	//base cases?
	//1) how will we know when we're at a deadend?
	//2) if we're on the exit, return true
	if( isExit(row, col) )
	{
		return( true );
	}
	
	//look around for valid in the 
	//order: up, right, down, left
	
	//if I can move up
		//move up and see if it leads to a solution
		//if so, return true
	
	//if I can move right
		//move right and see if it leads to a solution
		//if so, return true
		
	//repeat for remaining possible moves
	
	
	//What do I know if I didn't find a solution?
	//I know this position is NOT on a path 
	//to a solution
	
	//unmark this position
	//return false
}

File: eecs268/2025fall/MWF/2025.10.10.maze-walking-part-3/mazewalker.py
#mazewalker.py


class MazeWalker:
    def __init__(self, ???):
        self._maze = ???
        self._start_row = ???
        self._start_col = ???
        self._visited_grid = ???
        self._step_counter = 0

    def walk(self, row, col):
        self._mark(row, col)
        
        if self._is_exit(row, col):
            return True

        #Look for a valid move in the
        #order Up, Right, Down, Left

        #Up
        if self._is_valid_move(row-1, col):
            is_exit_found = self.walk(row-1, col)
            if is_exit_found:
                return True

        #Right
        if self._is_valid_move(row, col+1):
            is_exit_found = self.walk(row, col+1)
            if is_exit_found:
                return True
        #Down
        #Left
            
        #If we haven't found an exit,
        #out of options to try
        #Need to backtrack
        self._unmark(row, col)
        return False
    
    def _is_exit(self, row, col):
        #returns True if row,col is exit
        #returns False otherwise
